front/lib/production_checks/checks/scrub_deleted_core_document_versions.ts

```diff
@@ -161,32 +189,24 @@ async function scrubDocument({
   hasher.update(Buffer.from(document_id));
   const documentIdHash = hasher.digest("hex");
 
-  const path = `${dataSource.project}/${dataSource.internal_id}/${documentIdHash}/${hash}`;
+  // Legacy logic.
+  const legacyPath = `${dataSource.project}/${dataSource.internal_id}/${documentIdHash}/${hash}`;
 
-  const [files] = await storage
-    .bucket(DUST_DATA_SOURCES_BUCKET)
-    .getFiles({ prefix: path });
+  // New logic.
+  const path = `${dataSource.project}/${dataSource.internal_id}/${documentIdHash}`;
 
-  await Promise.all(
-    files.map((f) => {
-      if (!seen.has(f.name)) {
-        seen.add(f.name);
-        logger.info(
-          {
-            path: f.name,
-            documentId: document_id,
-            documentHash: hash,
-            dataSourceProject: dataSource.project,
-            dataSourceInternalId: dataSource.internal_id,
-            dataSourceId: dataSource.id,
-          },
-          "Scrubbing"
-        );
+  const bucket = storage.bucket(DUST_DATA_SOURCES_BUCKET);
 
-        return f.delete();
-      }
-    })
-  );
+  const localLogger = logger.child({
+    documentId: document_id,
+    documentHash: hash,
+    dataSourceProject: dataSource.project,
+    dataSourceInternalId: dataSource.internal_id,
+    dataSourceId: dataSource.id,
+  });
+
+  await deleteAllFilesFromFolder(localLogger, bucket, seen, path);
```

Comment:
`path` includes `legacyPath` (its parent) or that function deletes all childrens.

Out of clarity we want to keep both I agree for now, but we likely want to invert them otherwise the second call will be a no-op AFAICT

Reply:
Since you mentioned that we can't delete folders immediately, it wasn't going to delete the child folder. However, to be certain, I changed the order and added a comment.